packages/terra-status-view/src/StatusView.jsx

```diff
@@ -0,0 +1,211 @@
+  /**
+   * An array of actionable buttons to display.
+   */
+  children: PropTypes.node,
+
+  /**
+   *  Determines if the content should be aligned vertically at the top of the container rather than in the middle.
+   */
+  isAlignedTop: PropTypes.bool,
+
+  /**
+   * Determines if the glyph should be displayed.
+   */
+  isGlyphHidden: PropTypes.bool,
+
+  /**
+   * The descriptive text to display.
+   */
+  message: PropTypes.string,
+
+  /**
+   * The required status title to display. Status views with variants of type no-data, no-matching-results,
+   * not-authorized, and error will have defaulted values unless overridden with this prop.
+   */
+  title: PropTypes.string,
+
+  /**
+   * Determines the type of icon to display.
+   */
+  variant: PropTypes.oneOf(Object.values(variants)),
+};
+
+const defaultProps = {
+  children: [],
+  isAlignedTop: false,
+  isGlyphHidden: false,
+  message: undefined,
+  title: undefined,
+  variant: variants.ERROR,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class StatusView extends React.Component {
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      showGlyph: true,
+      paddingBottom: 0,
+      paddingTop: 0,
+    };
+    this.showAndCenterItems = this.showAndCenterItems.bind(this);
+  }
+
+  componentDidMount() {
+    // call after all items have rendered in the view to calculate which items
+    // can be shown and center the items inside the view appropriately
+    this.showAndCenterItems();
+  }
+
+  /**
+   * Will calculate which items inside the status view can appear based on fixed
+   * height of its container and will update the inner padding to center the items
+   * with 40% of the remaining space at the top with 60% on the bottom.
+   */
+  showAndCenterItems() {
+    const viewHeight = this.innerNode.offsetHeight;
+
+    let paddingTop = 0;
+    let paddingBottom = 0;
+    let showGlyph = false;
+
+    // get the content heights of the nodes that are to be shown if have content and cannot be hidden
+    let componentsHeight = this.titleNode.offsetHeight;
+    if (this.buttonsNode) {
+      componentsHeight += this.buttonsNode.offsetHeight;
+    }
+    if (this.messageNode) {
+      componentsHeight += this.messageNode.offsetHeight;
+    }
+    if (this.dividerNode) {
+      componentsHeight += this.dividerNode.offsetHeight;
+    }
+
+    if (this.glyphNode && viewHeight >= componentsHeight + this.glyphNode.offsetHeight) {
+      // if have a glyph to show and can fit inside the containers height, distribute remaining padding to center
+      showGlyph = true;
+      if (!this.props.isAlignedTop) {
+        const remainingHeight = viewHeight - (componentsHeight + this.glyphNode.offsetHeight);
+        paddingTop = remainingHeight * 0.4;
```

Comment:
Just curious, how did we decide on multiplying the padding with these values?

Reply:
it is specified in the callouts to distribute the remaining vertical space by 40% of it on top and 60% of the space below